conduit/data/datasets/audio/ecoacoustics.py

```diff
@@ -137,15 +123,16 @@ def __init__(
         super().__init__(x=x, y=y, transform=transform, audio_dir=self.base_dir)
 
     @property
-    def segment_len(self) -> Optional[float]:
+    def segment_len(self) -> float:
         return self._segment_len
 
     @segment_len.setter
-    def segment_len(self, value: Optional[float]) -> None:
+    def segment_len(self, value: float) -> None:
+        assert (
```

Comment:
raise a Value/Attribute error instead of asserting

Reply:
That said, rather than raising an error when the upper bound is violated, it might be better to simply cap the segment length.